### ISSUE TITLE
No module named ssl_match_hostname

### ISSUE THREAD

**Issue by mazz** [danschultzer/ansible-phoenix-build#1]
When I make a python27 virtualenv, try `source path/to/py27/bin/activate`, `pip install ansible`, `pip install docker`, `pip install docker-py` and then:

```ansible-playbook apps/build/update-docker-image.yml -vvv```

I get the error:

    "msg": "Failed to import docker-py - No module named ssl_match_hostname. Try `pip install docker-py`"

**Comment by danschultzer** [comment]
Uninstall `docker-py` and just have `docker` installed. `docker` is the newer version, and ansible will use it unless [`docker-py` is available](https://github.com/ansible/ansible/blob/e45c763b646e0609fb9c8d8b8cdf9c200985c313/lib/ansible/module_utils/docker_common.py#L41). I experienced a similar issue having both installed.

**Comment by danschultzer** [comment]
Let me know if that solves the issue @mazz, and I'll add it to the guide and repo as install instructions for `ansible`.

**Comment by mazz** [comment]
Hi I did
```pip uninstall docker-py```

and still got this message ...
```    "msg": "Failed to import docker-py - No module named ssl_match_hostname. Try `pip install docker-py`"```

**Comment by danschultzer** [comment]
Hmm, I'll try to replicate your setup when I got some time to figure out what goes wrong. What OS are you running on?

For now, could you try the following:

```bash
pip uninstall backports.ssl-match-hostname
pip install --upgrade docker
```

Run as `sudo` if necessary.

**Comment by mazz** [comment]
So I managed to get past this issue. My system python ansible needs to be ansible 2.2.1.0 for compatibility reasons for another project. To workaround this I did

virtualenv py27env
source py27env/bin/activate

I assumed from this point onward, that when I ran `pip` for example, I was running from the venv but that was not actually the case. I needed to refer to the relative path of the venv `pip` to install into the venv.
```
../../py27env/pip install -U pip
../../py27env/pip install ansible
../../py27env/ansible-playbook .ansible/apps/build/update-docker-image.yml -vvv
```
managed to solve it all(got to me an auth error but I thnk I just need to setup my user/pass
